api/v1/controllers/aspects.js

```diff
@@ -165,25 +167,26 @@ module.exports = {
     userProps.model.findAll(options)
     .then((usrs) => {
       users = usrs;
-      return u.findByKey(helper, params)
-      .then((o) => u.isWritable(req, o))
-        .then((o) => redisOps.getValue('aspect', o.name))
-        .then((cachedAspect) => {
-          if (cachedAspect) {
-            const userSet = new Set();
-            usrs.forEach((user) => {
-              userSet.add(user.dataValues.name);
-            });
-            cachedAspect.writers = cachedAspect.writers || [];
-            Array.from(userSet).forEach((user) => {
-              cachedAspect.writers.push(user);
-            });
-            return redisOps.hmSet('aspect', cachedAspect.name, cachedAspect);
-          }
-
-          // reject the promise if the aspect is not found in the cache
-          return Promise.reject(new apiErrors.ResourceNotFoundError());
+      return u.findByKey(helper, params);
+    })
+    .then((o) => u.isWritable(req, o))
+    .then((o) => redisOps.getHash('aspect', o.name))
```

Comment:
just curious why are we hashing this now ?

Reply:
In order to make all the ops methods work with the batching I had to enforce the rule that they have to return a single-stage promise (no multiple then chains). This was previously being done in two places: getValue and deleteSampleKeys. 

Those methods were deleted, and places that used them were updated to do multiple operations.

Previously getValue did getHash() then arrayObjsStringsToJson(). So here we do both of those directly, instead of calling getValue.